nemo/core/classes/modelPT.py

```diff
@@ -23,16 +24,39 @@
 
 from nemo.core import optim
 from nemo.core.classes.common import Model
+from nemo.core.config.base_config import Config
 from nemo.utils import logging
 
-__all__ = ['ModelPT']
+__all__ = ['ModelPT', 'ModelPTConfig']
+
+
+@dataclass
+class ModelPTConfig(Config):
+    """Inherit from this class when you parametrize NeMo models"""
+
+    optim: Optional[DictConfig] = None
+    train_ds: Optional[DictConfig] = None
+    validation_ds: Optional[DictConfig] = None
+    test_ds: Optional[DictConfig] = None
 
 
 class ModelPT(LightningModule, Model):
     """
     Interface for Pytorch-lightning based NeMo models
     """
 
+    def __init__(self, cfg: ModelPTConfig = None):
+        super().__init__()
+        self._train_dl = None
+        self._validation_dl = None
+        self._test_dl = None
+        self._optimizer = None
+        self._scheduler = None
+
+        if cfg is not None:
+            self.setup_training_data(cfg.train_ds)
```

Comment:
Maybe lets do a conditional check here to see if `cfg.train_ds` is None or not? Same for validation_ds below. Also, maybe test and add test_ds here too?

Reply:
done